File: Records-Store/API/Entities/Product.cs
namespace API.Entities
{
    public class Product
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Artist { get; set; }
        public long Price { get; set; }
        public string PictureUrl { get; set; }
        public string Genre { get; set; }
        public string Brand { get; set; }
        public int QuantityInStock { get; set; }
    }
}

File: Records-Store/client/src/app/models/product.ts
export interface Product {
    id: number;
    name: string;
    artist: string;
    price: number;
    pictureUrl: string;
    genre?: string;
    brand: string;
    quantityInStock?: number;
}

export interface ProductParams {
    orderBy: string;
    searchTerm?: string;
    genres: string[];
    brands: string[];
    pageNumber: number;
    pageSize: number;
}